lib/phoenix/verified_routes.ex

```diff
@@ -0,0 +1,757 @@
+defmodule Phoenix.VerifiedRoutes do
+  @moduledoc ~S"""
+  Provies route generation with compile-time verificiation.
+
+  Use of the `sigil_p` macro allows paths and URLs throughout your
+  application to be compile-time verified against your Phoenix router(s).
+  For example the following path and URL usages:
+
+
+      <.link href={~p"/sessions/new"} method="post">Sign in</.link>
+
+      redirect(to: url(~p"/posts/#{post}"))
+
+  Will be verified against your standard `Phoenix.Router` definitions:
+
+      get "/posts/:post_id", PostController, :show
+      post "/sessions/new", SessionController, :create
+
+  Unmatched routes will issue compiler warnings:
+
+      warning: no route path for AppWeb.Router matches "/postz/#{post}"
+        lib/app_web/controllers/post_controller.ex:100: AppWeb.PostController.show/2
+
+  Additionally, interpolated ~p values are encoded via the `Phoenix.Param` protocol.
+  For example, a `%Post{}` struct in your application may derive the `Phoenix.Param`
+  protocol to generate slug-based paths rather than ID based ones. This allows you to
+  use `~p"/posts/#{post}"` rather than `~p"/posts/#{post.id}"` throughout your
```

Comment:
To be coherent with the first part of the sentence, isn't `post.slug` instead of `post.id`?
```suggestion
  use `~p"/posts/#{post}"` rather than `~p"/posts/#{post.slug}"` throughout your
```